### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by adam-fowler**
Here's a starting point for the contributing document
Have I missed anything?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+## Submitting a Pull Request
+
+Please ensure to include the following in your Pull Request
+- A description of what you are trying to do. What the PR provides to the library, additional functionality, fixing a bug etc
+- A description of the code changes
+- Documentation on how these changes are being tested
+- Additional tests to show your code working and to ensure future changes don't break your code.
+
+Please keep your PRs to a minimal number of changes. If a PR is large try to split it up into smaller PRs. Don't move code around unnecessarily it makes comparing old with new very hard. If you have plans for a large change please talk to the maintainers of the project beforehand.
```
Comment (0xTim) [inline CONTRIBUTING.md:31]: We should mention the VSCode channel and nits

```suggestion
Please keep your PRs to a minimal number of changes. If a PR is large, try to split it up into smaller PRs. Don't move code around unnecessarily as it makes comparing old with new very hard. If you have plans for a large change please talk to the maintainers of the project beforehand. There is a `#vscode-swift` channel on the Swift Server Slack. You can [join here](https://join.slack.com/t/swift-server/shared_invite/zt-5jv0mzlu-1HnA~7cpjL6IfmZqd~yQ2A).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Contributing
+
+## Conduct
+All contributors are expected to adhere to the project's [Code of Conduct](CODE_OF_CONDUCT.md).
```
Comment (0xTim) [inline CONTRIBUTING.md:4]: Nit newline
```suggestion
## Conduct

All contributors are expected to adhere to the project's [Code of Conduct](CODE_OF_CONDUCT.md).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## Development
+
+To begin development on the VSCode extension for Swift you will need to install [Node.js](https://nodejs.org). On Linux, make sure to install Node.js from its official website or from [NodeSource](https://github.com/nodesource/distributions/) as the version included with your Linux distribution may be outdated.
+
+Next, clone this repository, and run `npm install` to install all the dependencies.
```
Comment (0xTim) [inline CONTRIBUTING.md]:
```suggestion
Next, clone this repository, and in the project directory run `npm install` to install all the dependencies.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+When you first open the project in VSCode you will be recommended to also install `ESLint` and `Prettier - Code formatter`. Please do so. We use these to ensure a consistent style and expect everyone who contributes to follow this style as well. 
+
+To run your version of the Swift extension while in VSCode, press `F5`. This will open up another instance of VSCode with it running. You can use the original version of VSCode to debug it. 
+
+## Submitting a bug or issue
+Please ensure to include the following in your bug report
```
Comment (0xTim) [inline CONTRIBUTING.md:17]: Nit spacing
```suggestion
## Submitting a bug or issue

Please ensure to include the following in your bug report:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+- A consise description of the issue, what happened and what you expected.
+- Simple reproduction steps
+- Version of the extension you are using
+- Contextual information (Platform, Swift version etc)
+
+## Submitting a Pull Request
+
+Please ensure to include the following in your Pull Request
```
Comment (0xTim) [inline CONTRIBUTING.md:25]:
```suggestion
Please ensure to include the following in your Pull Request:
```

**Review by 0xTim — CHANGES_REQUESTED** [review]
A few nits

**Review by 0xTim — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+All contributors are expected to adhere to the project's [Code of Conduct](CODE_OF_CONDUCT.md).
+
+## Development
+
+To begin development on the VSCode extension for Swift you will need to install [Node.js](https://nodejs.org). On Linux, make sure to install Node.js from its official website or from [NodeSource](https://github.com/nodesource/distributions/) as the version included with your Linux distribution may be outdated.
+
+Next, clone this repository, and in the project directory run `npm install` to install all the dependencies.
+
+When you first open the project in VSCode you will be recommended to also install `ESLint` and `Prettier - Code formatter`. Please do so. We use these to ensure a consistent style and expect everyone who contributes to follow this style as well.
```
Comment (svanimpe) [inline CONTRIBUTING.md:13]: There's a third one now: `connor4312.esbuild-problem-matchers`
It provides a problem matcher for ESBuild. Without it, you'll get an error when building.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+## Development
+
+To begin development on the VSCode extension for Swift you will need to install [Node.js](https://nodejs.org). On Linux, make sure to install Node.js from its official website or from [NodeSource](https://github.com/nodesource/distributions/) as the version included with your Linux distribution may be outdated.
+
+Next, clone this repository, and in the project directory run `npm install` to install all the dependencies.
+
+When you first open the project in VSCode you will be recommended to also install `ESLint` and `Prettier - Code formatter`. Please do so. We use these to ensure a consistent style and expect everyone who contributes to follow this style as well. 
+
+To run your version of the Swift extension while in VSCode, press `F5`. This will open up another instance of VSCode with it running. You can use the original version of VSCode to debug it. 
+
```
Comment (svanimpe) [inline CONTRIBUTING.md:16]: Can we also mention how to unit test?

Either:
- From VS Code, by selection "Extension Tests" in the Run and Debug activity.
- Using `npm run test` from the command line (when VS Code is not running, or you'll get an error)

File: CONTRIBUTING.md (hunk 6 above)
Comment (0xTim) [reply CONTRIBUTING.md:13]: Maybe we should just say "a number of Typescript extensions" to cover our bases and for any future additions/removals

Comment (adam-fowler) [reply CONTRIBUTING.md:13]: @svanimpe i don't have esbuild-problem-matchers installed and it seems to work ok. It was never suggested to me I should install it.

**Comment by 0xTim** [comment]
Spinning out the outstanding issues into #77 and #78 to merge for the release

**Pull request merged**